src/Utils/ArrayObject.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace PhpSchool\PhpWorkshop\Utils;
+
+use ArrayIterator;
+use IteratorAggregate;
+
+/**
+ * Class ArrayObject
+ * @package PhpSchool\PhpWorkshop\Utils
+ * @author  Aydin Hassan <aydin@hotmail.co.uk>
+ */
+class ArrayObject implements IteratorAggregate
+{
+
+    /**
+     * @var array
+     */
+    private $array;
+
+    /**
+     * @param array|mixed... $array
+     */
+    public function __construct($array)
+    {
+        if (func_num_args() > 1) {
+            $this->array = func_get_args();
+        } else {
+            $this->array = $array;
```

Comment:
Should this have an elseif `is_array` check then the else be `$this->array = [$array]` ?